### PR TITLE
CUSTCOM-13 A REST management DELETE command returns 415 code instead of 404

### PR DESCRIPTION
**Pull request by MattGill98**
This is a bug fix. Starting a fresh domain in 5.194 and running the following curl command:

~~~
curl -X DELETE -v -H "Content-Type: application/x-www-form-urlencoded" -H "X-Requested-By:Payara" http://localhost:4848/management/domain/servers/server/server/application-ref/non-existing-app-ref
~~~

will cause a HTTP 415 to return, where it should be a 404. Note that in 5.193 this worked.

# Testing

### New tests

Second commit. I've made the testing module fairly generic, as I see it as something that could be useful in the future.

### Testing Performed

The new test against 5.193 and the SNAPSHOT on micro, managed and remote profiles.

### Test suites executed
- Payara Samples (only single test)
- Java EE7 Samples (only JAX-RS tests against remote)
- Java EE8 Samples (only JAX-RS tests against remote)

# Notes for Reviewers
Read the commit messages ;)

I've deliberately ignored the previous fix that regressed. I hope this does the same job but with lower technical debt.

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins test please

**Comment by Cousjava** [comment]
Copyright notices need updating.

**Comment by MattGill98** [comment]
Good spot, I'm in 2019 mode still :wink:

**Comment by MattGill98** [comment]
jenkins test please

File: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/readers/FormReader.java [hunk 1]
```diff
@@ -64,22 +65,22 @@
...
 @Provider
-public class FormReader implements MessageBodyReader<HashMap<String, String>> {
+public class FormReader implements MessageBodyReader<Map<String, String>> {
     
     private static final String DEFAULT_CHARSET = StandardCharsets.UTF_8.toString();
 
     @Override
     public boolean isReadable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-        return type.equals(HashMap.class);
+        return type.isAssignableFrom(HashMap.class);
```
Comment (jbee) [inline nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/readers/FormReader.java]: Shouldn't it be `Map` now?

**Comment by dmatej** [comment]
Tested also:
`mvn clean verify -pl :payara-samples -amd -Ppayara-samples -fae`
- two tests are failing, but we know about them, they will be fixed in another issue.

**Comment by dmatej** [comment]
Jenkins test please

File: appserver/tests/payara-samples/samples/rest-management/src/test/java/fish/payara/samples/rest/management/HttpResponsesTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+@NotMicroCompatible
+public class HttpResponsesTest extends RestManagementTest {
+
+    /**
+     * Tests that when deleting a non-existent resource a 404 is returned instead of
+     * a 415. See CUSTCOM-13 for details.
+     */
+    @Test
+    public void when_DELETE_non_existent_resource_expect_404() {
```
Comment (dmatej) [inline appserver/tests/payara-samples/samples/rest-management/src/test/java/fish/payara/samples/rest/management/HttpResponsesTest.java]: camel case would be better, but ok.

Comment (MattGill98) [reply appserver/tests/payara-samples/samples/rest-management/src/test/java/fish/payara/samples/rest/management/HttpResponsesTest.java]: Noooo slaps on the wrist are owed for camel case in unit tests

**Pull request merged**